manifests/osfamily/darwin.pp

```diff
@@ -8,7 +8,6 @@
     $productversion_array = split($facts['os']['macosx']['version']['major'], '[.]')
     $productversion_major = $productversion_array[0]
   }
-
   if $puppet_agent::absolute_source {
```

Comment:
I think this whole block becomes a bit more readable and shorter, if you do something like:

```puppet
$source = if $puppet_agent::absolute_source {
...
```

Which enables you to get rid of all the following `$source =` in the logic block.